spec/symengine_spec.rb

```diff
@@ -18,5 +18,52 @@
     it { is_expected.to be_a SymEngine::Rational }
     its(:to_s) { is_expected.to eq '1/3' }
   end
+  
+  describe 'parse' do
+    subject { SymEngine::parse('123 + 321') }
+    
+    it { is_expected.to be_a SymEngine::Integer }
+    it { is_expected.to eq 444 }
+  end
+
+  it 'gives parse errors' do
+    expect { SymEngine::parse('12a + n34a9') }.to raise_error(RuntimeError)
```

Comment:
I partially agree to the above. While `RuntimeError` and `ZeroDivisionError` should be used, our code atleast shouldn't raise `NotImplementedError` and `SyntaxError`. The latter ones are `ScriptError`s and not `StandardError`s.
[This post](http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/) explains why it would be abuse of `NotImplementedError`. `SyntaxError` is not a right fit since it is raised while encountering invalid Ruby code. If the user wants to handle a parse error, (s)he will have to handle `SyntaxError` and that would also lead to unwanted silencing of the actual syntax error in the source code.

Also there is the technical problem that a `rescue` clause by default only catches the `StandardError`s. It is [highly advised that custom application level exceptions be derived from `StandardError`](http://blog.honeybadger.io/ruby-exception-vs-standarderror-whats-the-difference/).

For `SYMENGINE_UNDEFINED` it can be just `StandardError`.